plugins/uefi-capsule/fu-uefi-cod-device.c

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <fwupdplugin.h>
+
+#include "fu-uefi-cod-device.h"
+#include "fu-uefi-common.h"
+
+struct _FuUefiCodDevice
+{
+	FuUefiDevice		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuUefiCodDevice, fu_uefi_cod_device, FU_TYPE_UEFI_DEVICE)
+
+static gboolean
+fu_uefi_cod_device_get_results_for_idx (FuDevice *device, guint idx, GError **error)
+{
+	FuUefiDevice *device_uefi = FU_UEFI_DEVICE (device);
+	const gchar *tmp;
+	fwupd_guid_t guid = { 0x0 };
+	gsize bufsz = 0;
+	guint32 status = 0;
+	guint32 total_size = 0;
+	g_autofree gchar *err_msg = NULL;
+	g_autofree gchar *guidstr = NULL;
+	g_autofree gchar *name = NULL;
+	g_autofree gchar *version_str = NULL;
+	g_autofree guint8 *buf = NULL;
+
+	/* lets assume we just did one update */
+	name = g_strdup_printf ("Capsule%04u", idx);
+	if (!fu_efivar_get_data (FU_EFIVAR_GUID_EFI_GLOBAL, name,
+				 &buf, &bufsz, NULL, error)) {
```

Comment:
FU_EFIVAR_GUID_EFI_GLOBAL is not valid for Capsule#### we'll need EFI_CAPSULE_REPORT_GUID (from the EFI spec)

Reply:
Fixed, thanks.